src/routes/authentication/Authentication.component.tsx

```diff
@@ -115,36 +183,30 @@ const Authentication = () => {
             open: true,
             type: "success",
           });
-          setLoading(false);
-          navigate("/sign-in");
-          return;
+          user.authenticateUser(authDetails, {
+            onFailure(error) {
+              setSnackbarConfig({
+                message: handleSnackbarErrorMessage(error.message),
+                open: true,
+                type: "error",
+              });
+            },
+            onSuccess() {
+              Auth.currentAuthenticatedUser().then((res) => {
+                createUser(
+                  res.attributes.sub,
+                  res.signInUserSession.idToken.jwtToken
+                );
+              });
+            },
+          });
```

Comment:
This could all look much nicer if Cognito used promises rather than callbacks, but nothing we can do about that